extscanners/unwrap.go

```diff
@@ -0,0 +1,19 @@
+package extscanners
+
+import (
+	"github.com/efficientgo/core/errors"
+	"github.com/prometheus/prometheus/promql/parser"
+
+	"github.com/thanos-io/promql-engine/execution/parse"
+)
+
+func UnwrapConstVal(e parser.Expr) (float64, error) {
+	switch c := e.(type) {
+	case *parser.NumberLiteral:
+		return c.Val, nil
+	case *parser.StepInvariantExpr:
+		return UnwrapConstVal(c.Expr)
+	}
+
+	return 0, errors.Wrap(parse.ErrNotSupportedExpr, "argument must be a constant value")
```

Comment:
I would return regular error here and let the caller decide what to do with it i think

Reply:
Yeah, I created a new error so we can control on the callee level what we want to do with it. Let me know if you think that is better.